minvws/nl-kat-coordination
# Add 'set start date' functionality to scheduler

### Changes

Expose `deadline_at` field in the post request to the `schedules` endpoint. This allows rocky to set a 'start date' for a schedule to commence. In order words:

"I want this task to be recurring with this interval (cron) and we're going to start at 'start date'"

The `deadline_at` in the scheduler is used to push a task onto the queue at that moment: "the deadline". After that is done the next deadline is calculated based on the cron expression specified.

### QA notes

Not a notable change unless, the schedule at this date in the report scheduling of rocky is integrated.

Just make sure that all normal stuff continues to work. Let me know, and we can qa it together

---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
